middleware_auth_user.go

```diff
@@ -264,6 +267,14 @@ func (u *userMiddleware) getUserForOIDCHeader(ctx context.Context, header string
 	if err := token.Claims(&claims); err != nil {
 		return nil, errors.Wrap(err, "parsing token claims")
 	}
+	// if the subject starts with the spiffe route, then
+	// ignore it if it's the unauthorized user.
+	if strings.HasPrefix(token.Subject, spiffeRoute) {
+		// Istio-Ingressgateway-Public-Service-Account
+		if strings.Contains(token.Subject, unauthorizedSpifeServiceUser) {
```

Comment:
would we want to make sure it's a suffix? (with [HasSuffix](https://pkg.go.dev/strings#HasSuffix) or something)

Reply:
done